### PR TITLE
Introduce an "authselect enabled" rule

### PR DESCRIPTION
**Pull request by matejak**
The new rule features a check whether there is any authselect profile enabled. If it is not the case, it sets a profile according to an XCCDF value.

There are two test scenarios, the failing one is not remediated, as it is difficult to un-authselect a system - the authselect utility detects that there were previous modifications to other config files, and it won't apply the profile unless `--force` is used, which I wouldn't do in remediations.

### CODE REVIEW

File: ssg/build_yaml.py [hunk 1]
```diff
@@ -1128,7 +1128,7 @@ def to_xml_element(self, env_yaml=None):
         regex = (r'(package_.*_(installed|removed))|' +
                  r'(service_.*_(enabled|disabled))|' +
                  r'install_smartcard_packages$')
-        priority_order = ["installed", "install_smartcard_packages", "removed",
+        priority_order = ["enable_authselect", "installed", "install_smartcard_packages", "removed",
```
Comment (mildas) [inline ssg/build_yaml.py:1131]: Is this ordering correct? Do we really want it before `installed` rules?

Comment (yuumasato) [reply ssg/build_yaml.py:1131]: The priority ordering has effect within the group. Could it happen that `authselect` itself is not installed? 
This seems fine as the other package installed rule is `package_pam_pwquality_installed`.
https://github.com/ComplianceAsCode/content/tree/7ad2c970043370def2787a8ed773486384543821/linux_os/guide/system/accounts/accounts-pam

Comment (yuumasato) [reply ssg/build_yaml.py:1131]: The group `accounts-restrictions` also has rules remediating via `authselect`.

So I think we should guarantee that group `accounts-pam` is placed before group `accounts-restriction`  in https://github.com/ComplianceAsCode/content/blob/7ad2c970043370def2787a8ed773486384543821/ssg/build_yaml.py#L1159

Or we move the rule `authselect_enabled` one group up, into the `accounts` group.

File: linux_os/guide/system/accounts/accounts-pam/enable_authselect/rule.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+documentation_complete: true
+
+prodtype: rhel7,rhel8,ol8,rhel9,fedora
```
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-pam/enable_authselect/rule.yml:3]: Not necessary for rhel7.

File: linux_os/guide/system/accounts/accounts-pam/enable_authselect/tests/empty.fail.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# platform = multi_platform_rhel,multi_platform_fedora
+# remediation = none
+
+rm -rf /etc/authselect/*
+touch /etc/authselect/authselect.conf
```
Comment (yuumasato) [inline linux_os/guide/system/accounts/accounts-pam/enable_authselect/tests/empty.fail.sh:5]: I manually installed a RHEL8 and chose minimal install.
The installed system didn't have file authselect configued and file `/etc/authselect/authselect.conf` did not exist.
image

Does it make sense to add a test where only `/etc/authselect/authselect.conf` is missing?

Comment (marcusburghardt) [reply linux_os/guide/system/accounts/accounts-pam/enable_authselect/tests/empty.fail.sh]: At least in my researches I didn't find a way to "disable" `authselect` once a profile is selected unless the symbolic links are manually removed. If this is the case, the system could still have a content in `/etc/authselect/authselect.conf` but the authselect profile are not actually in use.

I would recommend to assess if an authselect profile is selected by testing if the PAM files are symlinks to /etc/authselect

Comment (matejak) [reply linux_os/guide/system/accounts/accounts-pam/enable_authselect/tests/empty.fail.sh]: Done, plus the remediation now remediates with `--force` if the RPM check of the `pam` package discovers no changes to the package - it is able to detect changes to config files. So if there are no changes, there is no reason why not to force.

**Comment by jan-cerny** [comment]
@matejak please rebase the PR on the latest upstream master so that the GitHub actions could execute the required CI job.
@vojtapolasek Is there anything else that needs to be done here?

**Comment by vojtapolasek** [comment]
I think it looks good. One question: Do you plan to add Ansible remediation as well?

**Comment by marcusburghardt** [comment]
> The issue could fix #8857.
> 
> I don't know how to write an Ansible remediation without an ugly shell code, so I would postpone this, as it can be done separately.

@matejak I can help with these Ansible remediation. I am actually working on the #8959 now and refactoring the bash macros for PAM. The next step will be the alignment between bash and Ansible macros, where I should work exclusively on Ansible improvements.

Finally, this PR plus the PR I am preparing for #8959 will fix the #8857 and help on the #8912.

File: linux_os/guide/system/accounts/enable_authselect/oval/shared.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    comment="Check that /etc/authselect/authselect.conf is not empty"
+    id="test_authselect_conf_not_empty" version="1">
+  <ind:object object_ref="obj_authselect_conf_not_empty" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="obj_authselect_conf_not_empty" version="1">
+    <ind:filepath>/etc/authselect/authselect.conf</ind:filepath>
+    <ind:pattern operation="pattern match">^.+$</ind:pattern>
+    <ind:instance operation="greater than or equal" datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
```
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/enable_authselect/oval/shared.xml:30]: Is this test still necessary? It is not used anyways.

File: linux_os/guide/system/accounts/enable_authselect/oval/shared.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    "password": "password-auth",
+    "postlogin": "postlogin",
+    "smartcard": "smartcard-auth",
+    "system": "system-auth",
+} -%}}
+
+<def-group>
+	<definition class="compliance" id="{{{ rule_id }}}" version="1">
+    {{{ oval_metadata("Check that authselect is enabled") }}}
+    <criteria operator="AND" comment="Check that /etc/authselect/authselect.conf is not empty">
```
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/enable_authselect/oval/shared.xml:12]: This `comment` text is not coherent to the criteria.

**Review by marcusburghardt — CHANGES_REQUESTED** [review]

**Comment by marcusburghardt** [comment]
All tests passed locally, in VM and Container:
```
Setting console output to log level INFO
INFO - The base image option has been specified, choosing Podman-based test environment.
INFO - Logging into logs_bash/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_enable_authselect
INFO - Script not_remediable.fail.sh using profile (all) OK
INFO - Script profile.pass.sh using profile (all) OK
INFO - Script remediable.fail.sh using profile (all) OK
```
```
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/mburghardt/OpenSCAP/Forks/ComplianceAsCode/content/logs/rule-custom-2022-06-24-1742/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_enable_authselect
INFO - Script not_remediable.fail.sh using profile (all) OK
INFO - Script profile.pass.sh using profile (all) OK
INFO - Script remediable.fail.sh using profile (all) OK
```

**Review by marcusburghardt — APPROVED** [review]
Thanks @matejak . This rule is pretty handy.

**Comment by marcusburghardt** [comment]
> Code Climate has analyzed commit [969d456](https://github.com/ComplianceAsCode/content/commit/969d456c255d0225ab5c7ee4a865dff58f858865) and detected **11 issues** on this pull request.
> 
> Here's the issue category breakdown:
> 
> Category	Count
> Style	1
> Bug Risk	10
> The test coverage on the diff in this pull request is **100.0%** (**50%** is the threshold).
> 
> This pull request will bring the total coverage in the repository to **42.7%** (**0.0%** change).
> 
> View more on [Code Climate](https://codeclimate.com/github/ComplianceAsCode/content/pull/8250).

Checked. We probably have to adjust the CodeClimate filters.

**Pull request merged**